src/test/java/htsjdk/samtools/filter/TagFilterTest.java

```diff
@@ -49,8 +48,34 @@
      * @param testValue         The value to test for in the record
      * @param expectedResult    The expected result (true is the sequence should match the filter, otherwise false)
      */
-    @Test(dataProvider="data")
-    public void testTagFilter(final String testName, final String tag, final List<Object> validValues,
+    @Test(dataProvider="dataInclude")
+    public void testIncludeTagFilter(final String testName, final String tag, final List<Object> validValues,
+                              final Object testValue, final boolean expectedResult, final boolean includeReads) {

```

Comment:
all the data provided by the dataprovider has `includeReads==true`. it would be good to copy over the (implicit) false test cases from below and have them here. Also, you could use a Boolean (rather than a boolean) and have a third value (null) which could imply using the `TagFilter(tag, validValues) constructor, thus refactoring even further the tests (since that the only difference between the two tests)